### PR TITLE
Backend/feat: #1274/clevertap-data-backend

### PR DESCRIPTION
**Pull request by prashant601**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [x] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/kafka-consumers/src/Consumer/BroadcastMessage/Processor.hs [hunk 1]
```diff
@@ -18,16 +18,16 @@ module Consumer.BroadcastMessage.Processor
...
+import "dynamic-offer-driver-app" Domain.Types.Message.MessageReport as Types
 import Environment
 import EulerHS.Prelude
 import qualified Kernel.External.Notification.FCM.Types as FCM
 import qualified Kernel.Storage.Esqueleto as Esq
 import Kernel.Types.Id
-import qualified Storage.Queries.Message.MessageReport as MRQuery
-import qualified Storage.Queries.Person as Person
-import Tools.Notifications (sendMessageToDriver)
+import "dynamic-offer-driver-app" Storage.Queries.Message.MessageReport as MRQuery
```
Comment (shanm-16) [inline Backend/app/kafka-consumers/src/Consumer/BroadcastMessage/Processor.hs:28]: why did you remove qualified namespace?

Comment (prashant601) [reply Backend/app/kafka-consumers/src/Consumer/BroadcastMessage/Processor.hs:28]: because before we were using just dynamic-offer-driver-app but now I have imported rider-app also in kafka consumer, so to not make these imports as ambiguous , I have used this syntax to import them

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/CleverTap.hs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import qualified Storage.Queries.Person.PersonStats as QPS
+import qualified Storage.Queries.Ride as QRide
+import qualified Storage.Queries.SavedReqLocation as QSRL
+import qualified Storage.Queries.SearchRequest as QSR
+import Tools.Error
+import Tools.Maps
+import Tools.Metrics (CoreMetrics)
+import qualified Tools.Whatsapp as WhatsApp
+
+data GetCleverTapDataRes = GetCleverTapDataRes
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/CleverTap.hs:40]: Rename everything from cleverTap to `personStats`

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs [hunk 3]
```diff
@@ -443,6 +448,8 @@ createPerson req mobileNumber notificationToken mbBundleVersion mbClientVersion
           flowStatus = DPFS.IDLE,
           updatedAt = person.updatedAt
         }
+    makePersonStats :: (EncFlow m r, EsqDBFlow m r, DB.EsqDBReplicaFlow m r, CacheFlow m r, CoreMetrics m) => SP.Person -> Id DMerchant.Merchant -> m DPS.PersonStats
+    makePersonStats person mId = SLP.backfillPersonStats person.id mId
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs:452]: Why do we need to backfill data for the newly created person? There will be no data for him, right?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Tabular/Person/PersonStats.hs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      completedRides Int
+      weekendRides Int
+      weekdayRides Int
+      offPeakRides Int
+      eveningPeakRides Int
+      morningPeakRides Int
+      weekendPeakRides Int
+      updatedAt UTCTime
+      Primary personId
+      UniquePersonStatsPersonId personId
```
[the remark this answers is not in the document]
Comment (hkmangla) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Tabular/Person/PersonStats.hs:46]: Not resolved yet.

[the remark this answers is not in the document]
Comment (prashant601) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Tabular/Person/PersonStats.hs]: done

File: Backend/app/kafka-consumers/src/Consumer/PersonStats/Processor.hs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  whenJust event.personId $ \pId -> do
+    when (event.service == RIDER_APP) do
+      let personId = Id pId
+          merchantId = Id event.merchantId
+      personStats <- Esq.runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+      when (isNotBackfilled personStats) do
+        personStatsValues <- SP.backfillPersonStats personId merchantId
+        Esq.runNoTransaction $ QP.incrementOrSetPersonStats personStatsValues
+      whenJust (event.payload) $ \payload -> do
+        when (event.eventType == RideEnded) $ do
```
[the remark this answers is not in the document]
Comment (prashant601) [reply Backend/app/kafka-consumers/src/Consumer/PersonStats/Processor.hs]: done

File: Backend/app/kafka-consumers/src/Consumer/PersonStats/Processor.hs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          when (payload.rs == DDR.COMPLETED) do
```
[the remark this answers is not in the document]
Comment (prashant601) [reply Backend/app/kafka-consumers/src/Consumer/PersonStats/Processor.hs]: done

File: Backend/app/kafka-consumers/src/Environment.hs [hunk 7]
```diff
@@ -27,7 +27,7 @@ import Kernel.Utils.App (lookupDeploymentVersion)
 import Kernel.Utils.Dhall
 import Kernel.Utils.IOLogging
 import Kernel.Utils.Servant.Client
-import Storage.CachedQueries.CacheConfig
+import "rider-app" Storage.CachedQueries.CacheConfig as CC
```
[the remark this answers is not in the document]
Comment (prashant601) [reply Backend/app/kafka-consumers/src/Environment.hs]: modified

File: Backend/app/rider-platform/rider-app/Main/src/API/UI.hs [hunk 8]
```diff
@@ -80,6 +81,7 @@ type API =
            :<|> Whatsapp.API
            :<|> Sos.API
            :<|> AppInstalls.API
+           :<|> CleverTap.API
```
[the remark this answers is not in the document]
Comment (prashant601) [reply Backend/app/rider-platform/rider-app/Main/src/API/UI.hs]: done

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs (hunk 3 above)
Comment (prashant601) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs]: modified .
just personstats needs to be created for new driver
